GetandTake/Core/Filters/LogActionFilterAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿using GetandTake.Configuration.Settings;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace GetandTake.Core.Filters;
+
+public class LogActionFilterAttribute : ResultFilterAttribute, IAsyncPageFilter
+{
+    private readonly ILogger<LogActionFilterAttribute> _logger;
+    private readonly AppSettings _appSettings;
+
+    public LogActionFilterAttribute(
+        ILogger<LogActionFilterAttribute> logger,
+        AppSettings appSettings)
+    {
+        _logger = logger;
+        _appSettings = appSettings;
+    }
+
+    public async Task OnPageHandlerExecutionAsync(
+        PageHandlerExecutingContext context,
+        PageHandlerExecutionDelegate next)
+    {
+        var page = context.RouteData.Values["page"];
+
+        if (_appSettings.LogFilter.IsLogFilterActive == true)
+        {
+            _logger.LogInformation(nameof(context), page);
+            await next.Invoke();
```

Comment:
Well, I am afraid if your condition will be `false` then this middleware chain method will not be executed. It might seriously affect your app workflow. Try to always put `await next.Invoke();` after some logic filter methods (at the end of the method I wanted to say)

Reply:
In this case: you mean that should be like that:
       
        if (_appSettings.LogFilter.IsLogFilterActive)
        {
            _logger.LogInformation(nameof(context), page);
        }
        await next.Invoke();

Reply:
Yes, this is correct :)